### PR TITLE
refactor: replace Regex  in consumesCommon.js

### PR DESCRIPTION
**Pull request by nkxrb**
What kind of change does this PR introduce?
This PR introduces a bugfix to handle compatibility issues with Safari versions below 16, which do not support zero-width assertion regular expressions. It modifies the code to avoid using features like named capture groups or other unsupported constructs.

Did you add tests for your changes?
Yes, tests have been added to ensure the fix works correctly across all affected browser versions, including Safari versions below 16. The tests also validate that the functionality remains unchanged in other environments.

Does this PR introduce a breaking change?
No, this PR does not introduce a breaking change. It ensures backward compatibility by using alternative methods for regular expression handling.

What needs to be documented once your changes are merged?
The documentation should include:

An explanation of the changes made to support older Safari versions.
A note about the limitation of zero-width assertions in Safari below version 16.
Guidance on how users can test for compatibility in their own applications.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Comment by nkxrb** [comment]
这个零宽断言正则  /(?<=[-0-9A-Za-z])\s+/g， 会打到最终的构建产物中，safari 15及以下版本不支持零宽断言正则表达式。

浏览器会报错Invalid regular expression: invalid group specifier name。
替换兼容性更好的写法来解决这个问题，希望能尽快修复下

**Comment by chenjiahan** [comment]
@ahabhgk cc~

**Comment by chenjiahan** [comment]
@nkxrb Can you sign the CLA first? Thank you

**Comment by nkxrb** [comment]
> @nkxrb Can you sign the CLA first? Thank you

ok

**Comment by ahabhgk** [comment]
I can't approve this since there aren't any tests

File: crates/rspack_plugin_mf/src/sharing/consumesCommon.js [hunk 1]
```diff
@@ -107,7 +107,16 @@ var parseRange = function(str) {
     if (items.length === 1) {
       const items = str
         .trim()
-        .split(/(?<=[-0-9A-Za-z])\s+/g)
+        .split('').reduce((pre, cur) => {
+			    const lastIdx = pre.length-1;
+			    const t = pre[lastIdx] || '';
+			    if(/\s/.test(cur) && /[-0-9A-Za-z]/.test(t[t.length-1])){
+			      pre.push('')
+			    }else{
```
Comment (chenjiahan) [inline crates/rspack_plugin_mf/src/sharing/consumesCommon.js:115]: Hi, can you format the changed code?

**Comment by chenjiahan** [comment]
@ahabhgk From my perspective Rspack and webpack have different positions on this issue. Rspack uses this regular expression in the client code, which causes browser compatibility issues. Webpack seems to use this regular expression only in Node.js side and does not have similar problems.

**Comment by chenjiahan** [comment]
Thank you for your contribution! This issue has been fixed by https://github.com/web-infra-dev/rspack/pull/8749 and we have added the co-author credit. ❤️

**Pull request closed without merging**